st2common/st2common/persistence/base.py

```diff
@@ -112,12 +112,52 @@ def aggregate(cls, *args, **kwargs):
         return cls._get_impl().aggregate(*args, **kwargs)
 
     @classmethod
-    def add_or_update(cls, model_object, publish=True, dispatch_trigger=True):
+    def insert(cls, model_object, publish=True, dispatch_trigger=True,
+               log_not_unique_error_as_debug=False):
+        if model_object.id:
+            raise ValueError('id for object %s was unexpected.' % model_object)
+        try:
+            model_object = cls._get_impl().insert(model_object)
+        except NotUniqueError as e:
+            if log_not_unique_error_as_debug:
+                LOG.debug('Conflict while trying to save in DB.', exc_info=True)
```

Comment:
I do agree that our logging pattern is pretty bogus through the code-base. Since we capture exceptions and raise new ones we completely loose context of the original execption and specifically the tarceback - while debugging issue this is like hitting a wall. Anyway, unless we can preserve the original traceback anc full exception content in the new exception I am reluctant to hide this exception completely.